Commonality Analysis Code_2.Rmd

```diff
@@ -205,18 +205,12 @@ regOut_dep_PP$APSRelatedMetrics
 
 boot.regOut_dep_PP <-boot(cope_ca_data_imputed_PP, boot.yhat, 1000, lmOut = depression_PP_imputed, regrout0 = regOut_dep_PP)
 boot.regOut_dep_PP_result <- booteval.yhat(regOut_dep_PP, boot.regOut_dep_PP, bty="perc")
-boot.regOut_dep_PP_result$combCIaps
 
-# Divide unique effects of hopelessness, agency, and PFS (the commonality coefficients) by R^2 of the model to get on a scale out of 100%.  We only need to do this for the confidence intervals.  We already have the main % total numbers from the previous step.
+#Total r^2 and CI outputs
+boot.regOut_dep_PP_result$combCIaps$R2[1:3]
```

Comment:
This is accurate. However, in general it's wise to stay away from "hard-coding" any references (e.g., referring to rows by number rather than by what you actually care about - the IV in that row). If for whatever reason you rearranged the order or IVs in your regression, this code would still work, but the results would suddenly be out of order - and you would have no way of knowing. 

This takes a little more space but it's safer. It can also replace the two lines below this by showing the `Commonality` and `R2` columns in one output.

```
boot.regOut_dep_PP_result$combCIaps %>%
  as_tibble(rownames = "IV") %>%
  filter(IV %in% c("residual_change_hopelessness", "residual_change_agency", "pi_pfs_mean")) %>%
  select(IV, Commonality, R2)
```

The `filter()` does the same thing that `[1:3]` does currently, but in a safer way (by referring to the IVs by name, not order).